koap/build.gradle.kts

```diff
@@ -58,3 +62,53 @@ kotlin {
         }
     }
 }
+
+// TODO: Put this in a separate file?
+fun String.runCommand(workingDir: File = file("./")): String {
+    val parts = this.split("\\s".toRegex())
+    val proc = ProcessBuilder(*parts.toTypedArray())
+            .directory(workingDir)
+            .redirectOutput(ProcessBuilder.Redirect.PIPE)
+            .redirectError(ProcessBuilder.Redirect.PIPE)
+            .start()
+    proc.waitFor(1, TimeUnit.MINUTES)
+    return proc.inputStream.bufferedReader().readText().trim()
+}
+
+tasks.withType<NpmPackageAssembleTask> {
+    // TODO: change to jsBrowserProductionWebpack for production
+    dependsOn("jsBrowserDevelopmentWebpack")
+}
+
+npmPublishing {
+    organization = "juullabs"
+    access = NpmAccess.RESTRICTED
+
+    repositories {
+        repository("github") {
+            access = NpmAccess.RESTRICTED
+            registry = uri("https://npm.pkg.github.com")
+        }
+    }
+
+    publications {
+        publication("${name}") {
+            moduleName = "${name}"
+            main = "${name}.js"
+
+            files {
+                from("${buildDir}/distributions")
+            }
+
+            dependencies {
+                npmPeer("kotlin", "*")
```

Comment:
This is kind of a workaround for being unable to forward on the Kotlin compiler/library version - asterisk means it can be whatever.

As a "peer" package, the consumer of this library will need to perform `npm install kotlin` in their project, and they can pick any version they want.

Reply:
This is a good note. Can you summarize and have it as a comment in the Gradle file, please?